### PR TITLE
BISERVER-12815 - Manage Data Sources: User can import an .xmi using 'Import Analysis'

### PR DESCRIPTION
**Pull request by Aliaksandr-Kastenka**
@rmansoor @lucboudreau 
Please review.

### CODE REVIEW

File: src/org/pentaho/platform/dataaccess/datasource/wizard/service/messages/messages.properties [hunk 1]
```diff
@@ -118,6 +118,8 @@ InMemoryDatasourceServiceImpl.ERROR_0020_UNABLE_TO_GET_DATASOURCE=Unable to cons
 
 ConnectionServiceHelper.ERROR_0001_UNABLE_TO_GET_CONNECTION_PASSWORD=Unable to get the connection password: {0}
 
+AnalysisService.ERROR_001_ANALYSIS_DATASOURCE_ERROR=Error creating Analysis datasource. Check the provided XML file.
```
Comment (lucboudreau) [inline src/org/pentaho/platform/dataaccess/datasource/wizard/service/messages/messages.properties]: This message should be more explicit. The only way it gets triggered is if the extension was .xmi. The message should reflect that.

**Comment by rmansoor** [comment]
@Aliaksandr-Kastenka Rather than not allowing only the .xmi file, we should try to not allow anything that is not a mondrian schema file. I was able to upload a log file as port of the mondrian schema upload.

File: src/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java [hunk 2]
```diff
@@ -379,6 +381,12 @@ protected void accessValidation() throws PentahoAccessControlException {
     super.validateAccess();
   }
 
+  protected void fileNameValidation( final String fileName ) throws PlatformImportException {
+    if ( fileName != null && fileName.endsWith( ".xmi" ) ) {
```
Comment (pentaho-nbaker) [inline src/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java]: Should be check that the filename ends with .xml?

**Comment by Aliaksandr-Kastenka** [comment]
@rmansoor @lucboudreau @pentaho-nbaker 
Pull-request updated. Please take a look.

**Pull request merged**